feat(trace): horizontal scroll on element

JonasBa opened the pull request:
Implements horizontal scroll on left column scrolling.

JonasBa committed:
feat(trace): add basic span bar rendering

JonasBa committed:
feat(trace): add column resizing

JonasBa committed:
feat(trace): add background color

JonasBa committed:
feat(trace): span rendering via matrix transform (#65065)

Computes span transformation using a matrix and enables us to define a
view later, effectively implementing horizontal zooming. Also adds min
scaling so we don't go under 1px in width spans

JonasBa committed:
feat(trace): add loading animation (#65052)

Not displaying anything was sad, so I added a simple animation that
indicates how the list will look like.

JonasBa committed:
Merge branch 'master' into jb/trace/span-bars-rendering

Abdkhan14 committed:
feature(new-trace-view): Rendering missing intrumentation bar. (#65099)

Couldn't fully replicate the styling for the bar as it renders in prod
without the bars getting janky. The stripes I added are a bit wider:
<img width="861" alt="Screenshot 2024-02-13 at 1 43 02 PM"
src="https://github.com/getsentry/sentry/assets/60121741/6067ebd6-792b-413b-b88c-9b441a1be720">
Updated color for autogrouped bars

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

JonasBa committed:
fxi(trace): nested autogroups (#65101)

Fix nested autogrouped cases

JonasBa committed:
feat(trace): scroll sync

JonasBa committed:
Merge branch 'master' into jb/trace/scroll-syncing

Abdkhan14 reviewed (approved):
Tested the functionality locally

Abdkhan14 commented on static/app/views/performance/newTraceDetails/virtualizedViewManager.tsx:
```diff
@@ -226,3 +284,47 @@ export class VirtualizedViewManager {
 }
+
+class RowMeasurer {
+  cache: Map<TraceTreeNode<any>, number> = new Map();
```
Why are we passing <any> here? Same on lines 296 and 317.

JonasBa committed:
feat(trace): render span duration (#65176)

Renders span durations on the timeline. 

There are 3 possible options
- the x coordinate is > half of column width, meaning it's on the right
side and it's label should be on the left
- the x coordinate is < less than half, meaning span is on the left and
its label should be to the right unless
the x coordinate + width are > half of column width, meaning the span
starts on the left, but spans a large part of the column width (at least
50%) and its label should be drawn on the inside

![CleanShot 2024-02-14 at 13 58
36@2x](https://github.com/getsentry/sentry/assets/9317857/59879049-aefd-4211-9d91-0725b175d02e)
